client/lib/features/admin/presentation/views/members_tab.dart

```diff
@@ -137,564 +189,220 @@ class _MembersTabState extends State<MembersTab> {
-                    color: context.theme.colorScheme.tertiaryFixed,
-                    size: 20,
-                  ),
-                ),
-              ),
-              Padding(
-                padding: EdgeInsets.symmetric(horizontal: 4),
-                child: ActionButton(
-                  height: 44,
-                  minWidth: 44,
-                  padding: EdgeInsets.zero,
-                  type: ActionButtonType.outline,
-                  shape: RoundedRectangleBorder(
-                    borderRadius: BorderRadius.circular(100),
-                  ),
-                  borderSide:
-                      BorderSide(color: context.theme.colorScheme.error),
-                  onPressed: () => alertOnError(
-                    context,
-                    () => _resolveRequest(request: request, approve: false),
-                  ),
-                  child: Icon(
-                    Icons.close,
-                    color: context.theme.colorScheme.error,
-                    size: 20,
-                  ),
-                ),
-              ),
-            ],
-          ),
-        ],
-      ),
-    );
-  }
-
-  Widget _buildRequestList(List<MembershipRequest> requestList) {
-    return ConstrainedBox(
-      constraints: BoxConstraints(maxHeight: 400, maxWidth: 600),
-      child: ListView(
-        shrinkWrap: true,
-        children: [
-          for (var i = 0; i < requestList.length; i++)
-            _buildRequestEntry(i, requestList[i]),
-        ],
-      ),
-    );
-  }
-
-  Widget _buildRequestsSection() {
-    return CustomStreamBuilder<List<MembershipRequest>>(
-      entryFrom: '_MembersTabState._buildRequestsSection',
-      stream: _requests,
-      errorMessage: 'Something went wrong loading requests. Please refresh.',
-      showLoading: false,
-      builder: (context, requestList) {
-        if ((requestList ?? []).isEmpty) {
-          return SizedBox.shrink();
+          );
+        } else if (membershipList == null) {
+          return Center(
+            child: CustomLoadingIndicator(),
+          );
         }
-        return Container(
-          padding: EdgeInsets.all(20),
-          decoration: BoxDecoration(
-            color: context.theme.colorScheme.surfaceContainerLowest,
-            borderRadius: BorderRadius.circular(10),
-          ),
-          child: Column(
-            crossAxisAlignment: CrossAxisAlignment.start,
-            children: [
-              HeightConstrainedText(
-                requestList!.isEmpty
-                    ? 'No Pending Join Requests'
-                    : 'Manage Join Requests (${requestList.length})',
-                style: AppTextStyle.headline4,
-              ),
-              SizedBox(height: 8),
-              if (requestList.isNotEmpty) _buildRequestList(requestList),
-            ],
-          ),
+        // There is a problem
+        return Center(
+          child: Text(context.l10n.errorLoadingMemberships),
```

Comment:
Maybe `throw` an error in the console too?

Reply:
We have a linter rule for `avoid_print`. Would you have an alternative suggestion?